specifyweb/frontend/js_src/lib/components/AppResources/TabDefinitions.tsx

```diff
@@ -97,29 +105,40 @@ const generateEditor = (xmlSpec: (() => BaseSpec<SimpleXmlNode>) | undefined) =>
     );
     const isReadOnly = React.useContext(ReadOnlyContext);
     return (
-      <CodeMirror
-        extensions={writable(extensions)}
-        readOnly={isReadOnly}
-        ref={handleRef}
-        theme={isDarkMode ? okaidia : xcodeLight}
-        onUpdate={({ state }): void => {
-          selectionRef.current = state.selection.toJSON();
-        }}
-        className={`w-full border border-brand-300 dark:border-none ${className}`}
-        /*
-         * Disable spell check if we are doing own validation as otherwise it's
-         * hard to differentiate between browser's spell check errors and our
-         * validation errors
-         */
-        spellCheck={typeof xmlSpec === 'function'}
-        value={data ?? ''}
-        /*
-         * FEATURE: provide supported attributes for autocomplete
-         *   https://codemirror.net/examples/autocompletion/
-         *   https://github.com/codemirror/lang-xml#api-reference
-         */
-        onChange={handleChange}
-      />
+      <div className="flex flex-col gap-2">
+        <div className="overflow-auto">
+          <CodeMirror
+            extensions={writable(extensions)}
+            readOnly={isReadOnly}
+            ref={handleRef}
+            theme={isDarkMode ? okaidia : xcodeLight}
+            onUpdate={({ state }): void => {
+              selectionRef.current = state.selection.toJSON();
+            }}
+            className={`w-full border border-brand-300 dark:border-none ${className}`}
+            /*
+             * Disable spell check if we are doing own validation as otherwise it's
+             * hard to differentiate between browser's spell check errors and our
+             * validation errors
+             */
+            spellCheck={typeof xmlSpec === 'function'}
+            value={data ?? ''}
+            /*
+             * FEATURE: provide supported attributes for autocomplete
+             *   https://codemirror.net/examples/autocompletion/
+             *   https://github.com/codemirror/lang-xml#api-reference
+             */
+            onChange={handleChange}
+          />
+        </div>
+        {showWarning &&
```

Comment:
I could see issues with this:
- when you are editing, warnings appear or disappear, causing the editor container height to change - does that happen? is that an issue?
- if there are a lot of warnings, this `<p>` becomes very tall and makes the codemirror container too small - you need to add some sort of max-height in here
- this has poor accessibility. better would be to put this in a ul with each error being a li and add `aria-live="polite"` to `ul` (read about what that does)